app.js

```diff
@@ -55,14 +56,15 @@ app.use(function(req, resp, next){
 app.use(reactRouted);
 app.use(express.static(__dirname + '/public', {maxAge: 3600000}));
 app.use(function(req, res, next) {
-  var firstPath = req.url.split("/")[1];
-  var search = req.search || "";
+  var location = url.parse(req.url).pathname;
+  var search = url.parse(req.url).search || "";
   // Get a valid locale from the path and header
-  var locale = getLocale(req.headers["accept-language"], firstPath);
-  // Compare the best valid locale to the path,
-  // if they don't match, redirect.
-  if (firstPath !== locale) {
-    res.redirect(301, "/" + locale + req.url + search);
+  var parsed = locationParser(req.headers["accept-language"], location);
+  var parsedLocale = parsed.locale;
+  var parsedRedirect = parsed.redirect;
+  // See if we should redirect.
+  if (parsedRedirect) {
```

Comment:
It was previously not happy with how confusing the `firstPath !== locale` in the old code was. It's much more clear now with `if (parsedRedirect)` is true, simply redirect using the parsed info.